.github/workflows/build-wheel-linux-arm64.yaml

```
@@ -37,291 +34,415 @@ jobs:
-            -v /var/run/docker.sock:/var/run/docker.sock \
-            -v `pwd`:/catalyst \
-            -i ${{ matrix.container_img }} \
-            bash /catalyst/.github/workflows/scripts/linux_arm64/rh8/build_mhlo.sh $GCC_VERSION ${{ matrix.python_version.major_minor }} ${{ matrix.python_version.patch }} ${{ matrix.python_version.package }}
-
-    - name: Save MHLO Build
-      id: save-mhlo-build
-      if: steps.cache-mhlo-build.outputs.cache-hit != 'true'
-      uses: actions/cache/save@v4
-      with:
-        path: ${{ github.workspace }}/mhlo-build
-        key: ${{ matrix.container_name }}-mhlo-${{ needs.constants.outputs.mhlo_version }}-wheel-build
-
-    - name: Build Enzyme
-      if: steps.cache-enzyme-build.outputs.cache-hit != 'true'
-      run: |
-        set -x
-        export GCC_VERSION=13
-        docker run --rm --platform linux/aarch64 \
-            -v /var/run/docker.sock:/var/run/docker.sock \
-            -v `pwd`:/catalyst \
-            -i ${{ matrix.container_img }} \
-            bash /catalyst/.github/workflows/scripts/linux_arm64/rh8/build_enzyme.sh $GCC_VERSION ${{ matrix.python_version.major_minor }} ${{ matrix.python_version.patch }} ${{ matrix.python_version.package }}
-
-    - name: Save Enzyme Build
-      id: save-enzyme-build
-      if: steps.cache-enzyme-build.outputs.cache-hit != 'true'
-      uses: actions/cache/save@v4
-      with:
-        path: ${{ github.workspace }}/enzyme-build
-        key: ${{ matrix.container_name }}-enzyme-${{ needs.constants.outputs.llvm_version }}-${{ needs.constants.outputs.enzyme_version }}-wheel-build
-
-  catalyst-linux-wheels-arm64:
+    steps:  
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: '20'
+
+      - name: Cache LLVM Source
+        id: cache-llvm-source
+        uses: actions/cache@v4
+        with:
+          path: ${{ github.workspace }}/mlir/llvm-project
+          key: llvm-${{ needs.constants.outputs.llvm_version }}-container-source
+          enableCrossOsArchive: True          
+
+      - name: Cache MHLO Source
+        id: cache-mhlo-source
+        uses: actions/cache@v4
+        with:
+          path: ${{ github.workspace }}/mlir/mlir-hlo
+          key: mhlo-${{ needs.constants.outputs.mhlo_version }}-container-source
+          enableCrossOsArchive: True  
+
+      - name: Cache Enzyme Source
+        id: cache-enzyme-source
+        uses: actions/cache@v4
+        with:
+          path: ${{ github.workspace }}/mlir/Enzyme
+          key: enzyme-${{ needs.constants.outputs.enzyme_version }}-container-source
+          enableCrossOsArchive: True
+
+      - name: Clone LLVM Submodule
+        if: steps.cache-llvm-source.outputs.cache-hit != 'true'
+        uses: actions/checkout@v4
+        with:
+          repository: llvm/llvm-project
+          ref: ${{ needs.constants.outputs.llvm_version }}
+          path: ${{ github.workspace }}/mlir/llvm-project
```

```suggestion
          path: mlir/llvm-project
```

Funny enough, `actions/checkout` will refuse to clone anything outside github.workspace so not sure why we even had that to begin with :sweat_smile: